tests/src/Interop/ICustomMarshaler/ICustomMarshaler.csproj

```diff
@@ -0,0 +1,34 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="12.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
+  <PropertyGroup>
+    <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
+    <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
+    <SchemaVersion>2.0</SchemaVersion>
+    <ProjectGuid>{95DFC527-4DC1-495E-97D7-E94EE1F7140D}</ProjectGuid>
+    <OutputType>Exe</OutputType>
+    <ProjectTypeGuids>{786C830F-07A1-408B-BD7F-6EE04809D6DB};{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}</ProjectTypeGuids>
+    <SolutionDir Condition="$(SolutionDir) == '' Or $(SolutionDir) == '*Undefined*'">..\..\</SolutionDir>
+    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+    <DefineConstants Condition="'$(TargetsWindows)' == 'true'">$(DefineConstants);Windows</DefineConstants>
```

Comment:
Hmm, any idea why this is being set on Unix CI (causing test fails)
```
#if Windows
        public const string LibcLibrary = "msvcrt.dll";
#else
        public const string LibcLibrary = "libc";
#endif
```

Reply:
I am unclear why this is occurring.

@RussKeldorph Can you think of any reason why the `Windows` define would be set on a Unix CI build? Based on `dirs.props`, the `TargetsWindows` property is only set on Windows. Thoughts?

Reply:
>#if Windows
>        public const string LibcLibrary = "msvcrt.dll";
>#else
>        public const string LibcLibrary = "libc";
>#endif

This is really weird. I added the code from above into `coreclr/tests/src/Interop/ICastable/Castable.cs` and built on MacOS. Using `ildasm`, the `"libc"` was used as expected.

Reply:
Can you try using the Is-Unix logic? I don't know if that works, but this could be a weird CI build issue.

Reply:
Hmmm, same thing - I am so confused!

Reply:
This should be addressed because tests are now being built on the platform the test will be running on.